### ISSUE TITLE
A way to specify offset/length when compressing and uncompressing

### ISSUE THREAD

**Issue by bgedik** [andrix/python-snappy#45]
If I want to compress or uncompress only a view of my string/data, I don't see a way to do that without creating a copy. Do you think this can be added.

**Comment by martindurant** [comment]
Have you considered using memoryview as a pure-python way to implement this?

**Comment by bgedik** [comment]
Yes. It gives the following error: `exceptions.TypeError: must be string or read-only buffer, not memoryview`

**Comment by martindurant** [comment]
I see, sorry to hear that.
Ideally it ought to work with memoryviews, but I see that the code is written in a pretty py2 kind of style.
Allowing an offset parameter doesn't sound too hard to implement, but I don't quite know how, I didn't write the original code. Perhaps an extra parameter to [here](https://github.com/andrix/python-snappy/blob/master/snappymodule.cc#L92)?
